test/CMakeLists.txt

```diff
@@ -28,6 +22,17 @@ if(LIBCORO_FEATURE_NETWORKING)
     )
 endif()
 
+if(LIBCORO_FEATURE_THREADING)
+    list(APPEND LIBCORO_TEST_SOURCE_FILES
+        bench.cpp
+        test_semaphore.cpp
+        test_io_scheduler.cpp
+        test_ring_buffer.cpp
+        test_shared_mutex.cpp
```

Comment:
This file probably cannot be moved easily, but perhaps the best we could do is maybe make the test that uses `coro::io_scheduler` be wrapped in a `#define` for the feature so not all of the tests are disabled?

Reply:
Fantastic, done!